lib/vault/api/sys/quota.rb

```diff
@@ -0,0 +1,75 @@
+module Vault
+  class Quota < Response
+    field :name
+    field :path
+    field :type
+  end
+
+  class RateLimitQuota < Quota
+    field :rate
+    field :burst
+  end
+
+  class LeaseCountQuota < Quota
+    field :counter
+    field :max_leases
+  end
+
+  class Sys
+    def quotas(type)
+      path = generate_path(type)
+      json = client.list(path)
+      if data = json.dig(:data, :key_info)
+        data.map do |item|
+          type_class(type).decode(item)
+        end
+      else
+        json
+      end
+    end
+
+    def create_quota(type, name, opts={})
+      path = generate_path(type, name)
+      client.post(path, JSON.fast_generate(opts))
+      return true
+    end
+
+    def delete_quota(type, name)
+      path = generate_path(type, name)
+      client.delete(path)
+      return true
+    end
+
+    def get_quota(type, name)
```

Comment:
Should we include a method for updating the config?

Reply:
Done